draft-aegis-mimi-arch-latest.md

```diff
@@ -429,6 +429,234 @@ TODO
 
 TODO
 
+# Managing Key Packages
+
+Each gateway implements a directory providing MLS key packages, which contain
+initial keying material of clients and can be used to add them to the group
+even when they are offline.
+
+Key packages are collected into _key pools_ of packages with common attributes.
+In particular, all key packages in a key pool belong to the same client and use
+the same cipher suite and protocol version. In addition, an application can
+define a custom label for key pools, for example, on that specifies custom MLS
+extensions clients must support.
+
+When a client is added to an MLS group (typically after being discovered as
+described in {{client-discovery}}), the application (or a gateway in case of
+federation) requests a key package from one of the client's pools with
+attributes matching those used by the group. By default, the returned key
+package is immediately deleted, in order to avoid key material reuse across
```

Comment:
Technically we don't immediately delete anymore, it is reserved to a single receiver until it is consumed. I can rework this language in a follow up though